crypto/hd/algorithm.go

```diff
@@ -1,96 +1,105 @@
 package hd
 
 import (
-	"fmt"
-
-	"github.com/pkg/errors"
-
 	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcutil/hdkeychain"
-	"github.com/tyler-smith/go-bip39"
+	bip39 "github.com/tyler-smith/go-bip39"
 
 	ethaccounts "github.com/ethereum/go-ethereum/accounts"
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 
-	tmcrypto "github.com/tendermint/tendermint/crypto"
-
-	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 
 	"github.com/cosmos/ethermint/crypto/ethsecp256k1"
 )
 
 const (
-	// EthSecp256k1 defines the ECDSA secp256k1 used on Ethereum
-	EthSecp256k1 = keys.SigningAlgo(ethsecp256k1.KeyType)
+	// EthSecp256k1Type defines the ECDSA secp256k1 used on Ethereum
+	EthSecp256k1Type = hd.PubKeyType(ethsecp256k1.KeyType)
 )
 
-// SupportedAlgorithms defines the list of signing algorithms used on Ethermint:
-//  - eth_secp256k1 (Ethereum)
-//  - secp256k1 (Tendermint)
-var SupportedAlgorithms = []keys.SigningAlgo{EthSecp256k1, keys.Secp256k1}
-
-// EthSecp256k1Options defines a keys options for the ethereum Secp256k1 curve.
-func EthSecp256k1Options() []keys.KeybaseOption {
-	return []keys.KeybaseOption{
-		keys.WithKeygenFunc(EthermintKeygenFunc),
-		keys.WithDeriveFunc(DeriveKey),
-		keys.WithSupportedAlgos(SupportedAlgorithms),
-		keys.WithSupportedAlgosLedger(SupportedAlgorithms),
-	}
-}
+var (
+	// SupportedAlgorithms defines the list of signing algorithms used on Ethermint:
+	//  - eth_secp256k1 (Ethereum)
+	//  - secp256k1 (Tendermint)
+	SupportedAlgorithms = keyring.SigningAlgoList{EthSecp256k1, hd.Secp256k1}
+	// SupportedAlgorithmsLedger defines the list of signing algorithms used on Ethermint for the Ledger device:
+	//  - eth_secp256k1 (Ethereum)
+	//  - secp256k1 (Tendermint)
+	SupportedAlgorithmsLedger = keyring.SigningAlgoList{EthSecp256k1, hd.Secp256k1}
```

Comment:
do we need a separate var for this? looks like its the same exact as `supportAlgorithms`. unless there's plans to add more to this that will differ from the above variable

Reply:
they could eventually differ. Ledger compatibility still needs to be tested